### DISCUSSION TITLE
v9.9

### DISCUSSION THREAD

**Discussion by Mizux** [google/or-tools#4119] [Announces]
# Dependencies

* abseil-cpp `20240116.1`.
* re2 `2024-02-01`.
* Protobuf `v25.3`.
* Scip `v810`.
* pybind11 `v2.11.1`.

* googletest `1.14.0`.

## C++ Changes

* use of `absl::Span` instead of `std::vector`.
* replace `const std::string&` with `absl::string_view`.
* use of absl log.

## CP-SAT

* Improvements to no_overlap_2d algorithms.
* rewrite python API to be pep8 compliant.  `model.AddBoolOr` -> `model.add_bool_or` 
* Improve the linear propagator code.
* Use the `~` operator in C++ and Python for the negation of a literal.

## Linear Solver

* make XPRESS support using a dynamic load (like gurobi support)

## Math Opt

* Sync with google3
* Add c++ and python support in CMake based build.
* Add math_opt in Python wheel package
* Enable few C++ tests with bazel based build.

## What's Changed
* build(deps): bump urllib3 from 2.0.4 to 2.0.7 in /bazel by @dependabot in https://github.com/google/or-tools/pull/3988
* Fix memory leaks in file::GetContents and file::SetContents by @pjh in https://github.com/google/or-tools/pull/4011
* build(deps): bump jupyter-server from 2.7.2 to 2.11.2 in /bazel by @dependabot in https://github.com/google/or-tools/pull/4014
* Improve the Xpress interface and add dynamic loading of the Xpress library by @sgatto in https://github.com/google/or-tools/pull/3966
* improve performance of Xpress interface and fix bug by @sgatto in https://github.com/google/or-tools/pull/4030
* build(deps): bump jinja2 from 3.1.2 to 3.1.3 in /bazel by @dependabot in https://github.com/google/or-tools/pull/4049
* Include LICENSE file in Python distribution by @chrisjbremner in https://github.com/google/or-tools/pull/4048

## New Contributors
* @pjh made their first contribution in https://github.com/google/or-tools/pull/4011
* @sgatto made their first contribution in https://github.com/google/or-tools/pull/3966
* @chrisjbremner made their first contribution in https://github.com/google/or-tools/pull/4048

**Full Changelog**: https://github.com/google/or-tools/compare/v9.8...v9.9

<hr /><em>This discussion was created from the release <a href='https://github.com/google/or-tools/releases/tag/v9.9'>v9.9</a>.</em>

**Comment by rauhs** [comment]
Hi. Will there be a NuGet for this?

**Reply by Mizux** [reply]
just uploaded ! https://www.nuget.org/packages/Google.OrTools/9.9.3963
